sklearn/tree/_criterion.pyx

```diff
@@ -599,6 +600,95 @@ cdef class Gini(ClassificationCriterion):
+
+    be the partial hellinger distance score for the k2 node.
+
+    Hellinger distance score is then defined as:
+
+        hellinger_distance = \sqrt{count_k1+count_k2}
+    """
+    cdef double node_impurity(self) nogil:
+        """Evaluate the impurity of the current node.
+        
+        Evaluate the Hellinger distance criterion as impurity of the current node,
+        i.e. the impurity of samples[start:end]. The smaller the impurity the
+        better.
+        """
+        cdef:
+            double impurity_left = 0.0
+            double impurity_right = 0.0
+            double impurity_sum = 0.0
+
+        # get the partial hellinger score, separately per child
+        self.children_impurity(&impurity_left, &impurity_right)
+
+        # hellinger score is sqrt of children impurity sum
+        impurity_sum = impurity_right + impurity_left
+        if impurity_sum >= 0:
+            impurity_sum = sqrt(impurity_sum)
+
+        return impurity_sum
+
+    cdef void children_impurity(self, double* impurity_left,
+                                double* impurity_right) nogil:
+        """Evaluate the impurity in children nodes.
+        
+        i.e. the impurity of the left child (samples[start:pos]) and the
+        impurity the right child (samples[pos:end]) using Hellinger distance.
+
+        Parameters
+        ----------
+        impurity_left : double pointer
+            The memory address to save the impurity of the left node to
+        impurity_right : double pointer
+            The memory address to save the impurity of the right node to
+        """
+        cdef:
+            # taking only first sum_left and sum_right because only single label is supported
+            double[::1] sum_left = self.sum_left[0]
+            double[::1] sum_right = self.sum_right[0]
+
+            # sum of same class in both children nodes
+            double sum_k1 = sum_left[0] + sum_right[0]
+            double sum_k2 = sum_left[1] + sum_right[1]
+
+            # hellinger distance score calculation helpers
+            double count_k1_left = 0.0
+            double count_k2_left = 0.0
+            double count_k1_right = 0.0
+            double count_k2_right = 0.0
+
+        # in case any of the classes sum is 0 it means that
+        # the classes are perfectly separated and no additional split is required
+        if  sum_k1 > 0 and sum_k2 > 0:
+            count_k1_left = sqrt(sum_left[0] / sum_k1)
+            count_k1_right = sqrt(sum_right[0] / sum_k1)
+            count_k2_left = sqrt(sum_left[1] / sum_k2)
+            count_k2_right = sqrt(sum_right[1] / sum_k2)
+
+        # assigning only first impurity because only single label is supported
+        # hellinger distance score is, the higher, the better
+        # split decision is, the lower, the better
+        # thus adding '1 - score' in order to adapt hellinger distance score to split mechanism
```

Comment:
```suggestion
        # Assigning only the positive class impurity as the negative class 
        # impurity can be inferred in binary classification.
        # 
        # The higher Hellinger distance, the better, whilst for split decision,
        # the lower the score, the better. Thus we compose the Hellinger distance
        # with 'x ↦ 1 - x' in order to adapt it for the split decision.
```

Reply:
rephrased the second part per your suggestion

regarding the first part, `assigning only first impurity because only single label is supported`, this comment of mine I think is redundant and in your suggestion I'm not clear about the classes, because I do use both negative and positive class for score calculation. Suggest to remove my original comment, I think it was a mistake